61.md

```diff
@@ -0,0 +1,36 @@
+NIP-61
+======
+
+Danmaku
+-------
+
+`draft` `optional`
+
+This NIP defines a new kind of event, `2333`, for danmaku.
+
+Danmaku (弾幕 だんまく; 弹幕; literally translated as "bullet curtain" and figuratively as "barrage") is a subtitle system in online video platforms that originates from Japan and popularised in Mainland China. Such system allows user to post moving comments onto a playing video that are synchronized to the video timeline. The comments are typically presented as “shooting” across the screen, resembling a barrage.
+
+### Format
+
+The format uses a regular event kind `2333`.
+
+The `.content` of these events is the content of the danmaku comment.
+
+The list of tags are as follows:
+* `i` (required) universally unique identifier of the video. The format is `<platform (lowercase)>:<identity>` where `<platform>` is the platform of the video and `<identity>` is the unique identifier of the video on that platform. For example, a YouTube video with the identifier `aA-jiiepOrE` would have the identifier `youtube:aA-jiiepOrE`.
```

Comment:
Why not just use a `r` with the url of the video instead of inventing our own new data format? If we're talking podcast episode guids or something that's different, but for referring to centralized entities where the client is bundled with the data, I'd just use a url.

Reply:
Because the URL for the same video can vary with different parameters, and sometimes unexpected ones at that. The URL for the same video can be different on mobile and desktop platforms, and also differ between China and the US. So, I think having a simple ID convention would be better. If someone wants to use the URL directly, that's also possible because it still conforms to the <a>:<b> format.

Reply:
Interesting. I feel like this points to a bigger problem with standardization; regular urls basically treat closed services like protocols, which you're just making explicit here. I'm not sure how to solve this long-term.

Reply:
I believe platforms are unlikely to change IDs easily, so for now, using IDs seems to be a better approach than using URLs. Indeed, this cannot completely solve the problem.